### PR TITLE
use ensurepip, other small cleanups

### PR DESCRIPTION
**Pull request by mattip**
- document quay.io/organization/pypa for easy reference
- remove 3.4 from the documents
- replace get-pip with ensurepip
- add a hash for wheel
- pin python2.7 requirements since setuptools 46 doesn't support it xref gh-491)

### CODE REVIEW

File: docker/build_scripts/build_utils.sh [hunk 1]
```diff
@@ -54,14 +54,18 @@ function do_cpython_build {
     if [ -e ${prefix}/bin/python3 ]; then
         ln -s python3 ${prefix}/bin/python
     fi
-    ${prefix}/bin/python get-pip.py
+    ${prefix}/bin/python -mensurepip
```
Comment (auvipy) [inline docker/build_scripts/build_utils.sh]: -m ensurepip?

Comment (trishankatdatadog) [reply docker/build_scripts/build_utils.sh]: Good catch! Was this PR tested?

Comment (mattip) [reply docker/build_scripts/build_utils.sh]: why? Both `-mensurepip` and `-m ensurepip` work.

Comment (trishankatdatadog) [reply docker/build_scripts/build_utils.sh]: I see. Yes, but the second one is more readable.

Comment (mattip) [reply docker/build_scripts/build_utils.sh]: changing, thanks

**Comment by mattip** [comment]
What is the proper git incantation to merge this to manylinux2014? Do we merge the branch into manylinux2014 or cherry-pick the commits?

**Comment by mayeut** [comment]
@mattip, that would be cherry-picking the commits.

**Pull request merged**